### PR TITLE
Refactor: instance list to handle PAYG instance

### PR DESCRIPTION
**Pull request by 1yam**
Problem:
```aleph instance list``` only handles the Hold method.

Solution:
Check the payment type and fetch api_url and ipv6 from ```/about/executions/list.```

Changes:
- Added logic to check if there is a payment associated with the message.
- If there is no payment, the function fetches the VM details from the scheduler API.
- If there is a payment, the function fetches the server URL and retrieves the ipv6 address from the node's execution list

### CODE REVIEW

**Comment by Psycojoker** [comment]
@1yam can you rebase this branch when you have some time plz? It conflicts with master now

File: src/aleph_client/commands/instance.py [hunk 1]
```diff
@@ -256,20 +256,46 @@ async def delete(
...
+                        ]
+                        return message.item_hash, ipv6_address
+
+            return message.item_hash, "Not available (yet)"
         except ClientResponseError:
             return message.item_hash, "Not available (yet)"
 
 
-async def _show_instances(messages: List[InstanceMessage]):
+async def _show_instances(messages: List[InstanceMessage], node_list):
```
Comment (Psycojoker) [inline src/aleph_client/commands/instance.py:298]: Can you add some typing to `node_list` plz?

File: src/aleph_client/commands/instance.py [hunk 2]
```diff
@@ -256,20 +256,46 @@ async def delete(
...
 
 
-async def _get_ipv6_address(message: InstanceMessage) -> Tuple[str, str]:
+async def fetch_json(session: ClientSession, url: str) -> dict:
+    async with session.get(url) as resp:
+        resp.raise_for_status()
+        return await resp.json()
+
+
+async def _get_ipv6_address(message: InstanceMessage, node_list) -> Tuple[str, str]:
```
Comment (Psycojoker) [inline src/aleph_client/commands/instance.py:265]: Can you add some typing to `node_list` plz?

**Review by Psycojoker — CHANGES_REQUESTED** [review]

**Comment by 1yam** [comment]
Clean PR : [](https://github.com/aleph-im/aleph-client/pull/224)

**Pull request closed without merging**